config/common.go

```diff
@@ -34,6 +34,14 @@ type Unmarshallable interface {
 	Unmarshal(component *confmap.Conf) error
 }
 
+// Marshallable defines an optional interface for custom configuration marshalling.
+// A configuration struct can implement this interface to override the default marshalling.
+type Marshallable interface {
```

Comment:
Let's have this in `confmap`, I think in a followup PR we can have the `Unmarshallable` also moved to confmap and deprecated from here.

Reply:
Moved it under `confmap`.